packages/event-storage-adapter-http/src/adapter.ts

```diff
@@ -0,0 +1,82 @@
+import uniqBy from 'lodash/uniqBy';
+import { OpenAPI } from 'openapi-types';
+
+import {
+  EventDetail,
+  EventStorageAdapter,
+  GroupedEvent,
+  ListAggregateIdsOutput,
+} from '@castore/core';
+
+import { getApiMethod, getMethodFromPath } from './utils/getApiMethod';
+import { getSwaggerClient } from './utils/getSwaggerClient';
+import { ApiResponse, SwaggerClient, Path } from './utils/types';
+
+export class HttpEventStorageAdapter implements EventStorageAdapter {
+  getEvents: EventStorageAdapter['getEvents'];
+  listAggregateIds: EventStorageAdapter['listAggregateIds'];
+  pushEvent: EventStorageAdapter['pushEvent'];
+  pushEventGroup: EventStorageAdapter['pushEventGroup'];
+  groupEvent: EventStorageAdapter['groupEvent'];
+  swagger: OpenAPI.Document;
+  client: Promise<SwaggerClient>;
+  getApiMethod: (method: string) => Path | undefined;
+
+  constructor({ swagger }: { swagger: OpenAPI.Document }) {
+    this.swagger = swagger;
+    // @ts-ignore the constructor returns a promise
+    this.client = getSwaggerClient({ spec: this.swagger });
+    this.getApiMethod = getApiMethod(this.swagger);
```

Comment:
Y a moyen d'être un peu moins ambigu dans le naming. `this.getOperation = getOperationResolver(this.swagger)` ?